sentry_sdk/integrations/spark/spark_driver.py

```diff
@@ -0,0 +1,191 @@
+from sentry_sdk import configure_scope
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import Integration
+
+
+class SparkIntegration(Integration):
+    identifier = "spark"
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        patch_spark_context_init()
+
+
+def _set_app_properties():
+    from pyspark import SparkContext
+
+    sparkContext = SparkContext._active_spark_context
+    if sparkContext:
+        sparkContext.setLocalProperty("app_name", sparkContext.appName)
```

Comment:
Can we safely do this without interfering with user-set properties? Why do we need this?

Reply:
We set a local property so that the worker has access to "app_name" and "application_id".

```python
if "app_name" in taskContext._localProperties:
    scope.set_tag("app_name", taskContext._localProperties["app_name"])
    scope.set_tag(
        "application_id", taskContext._localProperties["application_id"]
    )
```

It might be a good idea to rename it to something more sentry specific so that it won't get changed by the user. `sentry_integration_app_name`?

Reply:
I don't understand why this is necessary. Couldn't you set them on the scope directly?

(also I would prefer it if you used an event processor here like in the other code sample I posted)

Reply:
https://github.com/apache/spark/blob/master/python/pyspark/context.py#L1015-L1020

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L668-L685

The driver propagates these local properties to the workers, I'm essentially using this as a way to communicate between the driver process and worker process (because they are on different clusters). We set the local property with the driver integration so that the worker integration can access it.

If we don't use `setLocalProperty` in the driver, the worker integration cannot set tags for `application_id` and `app_name` (which makes it much harder to pair worker errors with driver errors by tag in Sentry itself). I have not found any other way to access this information with a worker.

Did change to use an event processor 👍

Reply:
Makes sense!

Reply:
Hmm, could you add a short comment that this is how you transport data over the wire